$OpenBSD: patch-main_c,v 1.1 2011/04/20 20:51:41 naddy Exp $

Fix Savannah bug #30723: expand MAKEFLAGS before we re-exec after
rebuilding makefiles.

--- main.c.orig	Sun Apr 17 16:26:49 2011
+++ main.c	Sun Apr 17 16:27:56 2011
@@ -2093,6 +2093,7 @@ main (int argc, char **argv, char **envp)
             const char *pv = define_makeflags (1, 1);
             char *p = alloca (sizeof ("MAKEFLAGS=") + strlen (pv) + 1);
             sprintf (p, "MAKEFLAGS=%s", pv);
+            p = allocated_variable_expand (p);
             putenv (p);
           }
 
